File: backend/internal/model/zenn.go
package model

import "time"

// ZennArticle はZennから同期した記事データを表す。
type ZennArticle struct {
	ID            uint      `json:"id"`
	UserID        uint      `json:"user_id"`
	ZennID        int64     `json:"zenn_id"` // Zenn側の記事ID
	Title         string    `json:"title"`
	Slug          string    `json:"slug"`         // URL用のスラッグ
	Emoji         string    `json:"emoji"`        // 記事のアイコン絵文字
	ArticleType   string    `json:"article_type"` // "tech"（技術記事）または "idea"（アイデア記事）
	LikedCount    int       `json:"liked_count"`
	CommentsCount int       `json:"comments_count"`
	PublishedAt   time.Time `json:"published_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

// ZennStats はユーザーのZenn記事統計情報を表す。
// DBテーブルには対応せず、集計結果を格納するDTO。
type ZennStats struct {
	TotalArticles int `json:"total_articles"`
	TotalLikes    int `json:"total_likes"`
	TotalComments int `json:"total_comments"`
}

File: frontend/src/api/zenn.ts
import client from './client';

export interface ZennArticle {
  id: number;
  user_id: number;
  zenn_id: number;
  title: string;
  slug: string;
  emoji: string;
  article_type: string;
  liked_count: number;
  comments_count: number;
  published_at: string;
  updated_at: string;
}

export interface ZennStats {
  total_articles: number;
  total_likes: number;
  total_comments: number;
}

export const connectZenn = (username: string) =>
  client.post('/zenn/connect', { username });

export const disconnectZenn = () =>
  client.delete('/zenn/disconnect');

export const syncZenn = () =>
  client.post('/zenn/sync');

export const getZennArticles = (userId: number) =>
  client.get<ZennArticle[]>(`/zenn/articles/${userId}`);

export const getZennStats = (userId: number) =>
  client.get<ZennStats>(`/zenn/stats/${userId}`);
